### PR TITLE
[6.2.z][Chery-pick]-CLI-API contentview version removal (stubed + imlementation) + conflict resolution

### PR DESCRIPTION
**Pull request by ldjebran**
```console
dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_contentview.py -v -k "test_positive_delete_cv_promoted_to_multi_env or test_positive_remove_cv_version_from_env or test_positive_remove_cv_version_from_multi_env or test_positive_remove_prod_promoted_cv_version_from_default_env or test_positive_remove_promoted_cv_version_from_default_env or test_positive_remove_qe_promoted_cv_version_from_default_env or test_positive_remove_renamed_cv_version_from_default_env"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /home/dlezz/bin/redhat/6.2.z/python/2.7/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: xdist-1.14, cov-2.3.1
collected 59 items 

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_delete_cv_promoted_to_multi_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_env_with_host_registered <- ../../bin/redhat/6.2.z/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_multi_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_multi_env_capsule_scenario <- ../../bin/redhat/6.2.z/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_prod_promoted_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_promoted_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_qe_promoted_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_renamed_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED

 50 tests deselected by '-ktest_positive_delete_cv_promoted_to_multi_env or test_positive_remove_cv_version_from_env or test_positive_remove_cv_version_from_multi_env or test_positive_remove_prod_promoted_cv_version_from_default_env or test_positive_remove_promoted_cv_version_from_default_env or test_positive_remove_qe_promoted_cv_version_from_default_env or test_positive_remove_renamed_cv_version_from_default_env' 
================================ 7 passed, 2 skipped, 50 deselected in 2040.63 seconds =================================


dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/api/test_contentviewversion.py -v -k "test_positive_remove_cv_version_from_multi_env or test_positive_remove_cv_version_from_env or test_positive_remove_prod_promoted_cv_version_from_default_env or test_positive_remove_qe_promoted_cv_version_from_default_env or test_positive_remove_renamed_cv_version_from_default_env or test_positive_delete_cv_promoted_to_multi_env"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /home/dlezz/bin/redhat/6.2.z/python/2.7/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: xdist-1.14, cov-2.3.1
collected 19 items 

tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_delete_cv_promoted_to_multi_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_cv_version_from_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_cv_version_from_env_with_host_registered <- ../../bin/redhat/6.2.z/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_cv_version_from_multi_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_cv_version_from_multi_env_capsule_scenario <- ../../bin/redhat/6.2.z/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_prod_promoted_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_qe_promoted_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentviewversion.py::ContentViewVersionDeleteTestCase::test_positive_remove_renamed_cv_version_from_default_env <- robottelo/decorators/__init__.py PASSED

 11 tests deselected by '-ktest_positive_remove_cv_version_from_multi_env or test_positive_remove_cv_version_from_env or test_positive_remove_prod_promoted_cv_version_from_default_env or test_positive_remove_qe_promoted_cv_version_from_default_env or test_positive_remove_renamed_cv_version_from_default_env or test_positive_delete_cv_promoted_to_multi_env' 
================================ 6 passed, 2 skipped, 11 deselected in 1023.40 seconds =================================
```

### CODE REVIEW

File: tests/foreman/api/test_contentviewversion.py [hunk 1]
```diff
@@ -310,6 +320,595 @@ def test_negative_delete(self):
...
+
+        @id: 7d5961d0-6a9a-4610-979e-cbc4ddbc50ca
+
+        @Steps:
+
+        1. Create a content view
+        2. Add a yum repo to the content view
+        3. Publish the content view
+        4. Rename the content view
+        5. remove the published version from Library environment
```
Comment (pondrejk) [inline tests/foreman/api/test_contentviewversion.py]: You start the last step with lower case in several docstrings, sorry for the nitpick but the docs person in me cannot unsee it :)

File: tests/foreman/api/test_contentviewversion.py [hunk 2]
```diff
@@ -310,6 +320,595 @@ def test_negative_delete(self):
...
+        content_view.delete_from_environment(lce_library.id)
+        # assert that the content view version does not exist in Library
+        # environment
+        self.assertEqual(len(content_view_version.read().environment), 0)
+
+    @run_only_on('sat')
+    @tier2
+    def test_positive_remove_qe_promoted_cv_version_from_default_env(self):
+        """Remove QE promoted content view version from Library environment
+
```
Comment (pondrejk) [inline tests/foreman/api/test_contentviewversion.py]: As CV names are arbitrary, I'm not sure if we should include them in test names. I think the distinctive thing about this test is that it syncs docker content, I'd mention it at least in the description. How about test_positive_remove_promoted_docker_cv_version_from_default_env?

File: tests/foreman/api/test_contentviewversion.py [hunk 3]
```diff
@@ -310,6 +320,595 @@ def test_negative_delete(self):
...
+        # assert that the content view version does not exist in Library
+        # environment
+        self.assertEqual(
+            {lce_dev.id, lce_qe.id},
+            {lce.id for lce in content_view_version.read().environment}
+        )
+
+    @run_only_on('sat')
+    @tier2
+    def test_positive_remove_prod_promoted_cv_version_from_default_env(self):
```
Comment (pondrejk) [inline tests/foreman/api/test_contentviewversion.py]: see the comment about cv names above

**Comment by ldjebran** [comment]
@pondrejk, thanks your comments, this is a cherry pick of already existing tests on master,
I will record your comments, so that they can be realized in a refactoring PR, that should be cherry-picked to master also after merge.

**Comment by ldjebran** [comment]
@pondrejk , I have make some changes only to manage some conflicts with the current branch

**Comment by pondrejk** [comment]
imho, it is not significant if a patch is applied first in master or vice versa, but I'm not going to block the party, so ACK

**Pull request merged**